numpy/distutils/fcompiler/gnu.py

```diff
@@ -126,8 +126,10 @@ def get_flags_linker_so(self):
                 # from it.
                 import distutils.sysconfig as sc
                 g = {}
-                filename = sc.get_makefile_filename()
-                sc.parse_makefile(filename, g)
+                if '__pypy__' not in sys.builtin_module.names:
```

Comment:
I feel like a try/except might be more appropriate than an explicit pypy check? That way we're robust against pypy adding a makefile, or being used with some other python implementation that also doesn't have a makefile.